.github/workflows/main.yml

```diff
@@ -466,3 +466,84 @@ jobs:
           path: |-
             ${{ github.workspace }}/aws-cdk/dist/
             ${{ github.workspace }}/aws-cdk/**/dist/
+  push-go-runtime:
+    needs: build
+    if: github.event_name == 'push'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Node 12
+        uses: actions/setup-node@v2.1.4
+        with:
+          node-version: '12'
+      - name: Download Artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: built-tree
+          path: ${{ github.workspace }}/jsii-built-tree
+      - name: Extract Artifact
+        working-directory: ${{ github.workspace }}/jsii-built-tree
+        run: |-
+          echo "::group::Untar Archive"
+          tar Jxvf built-tree.tar.xz
+          echo "::endgroup"
+
+          rm built-tree.tar.xz
+      - name: Get Last Commit Message
+        id: get-last-commit-msg
+        working-directory: ${{ github.workspace }}/jsii-built-tree
+        run: |-
+          message=$(git log -1 --format=%s)
+          echo "::set-output name=result::$message"
+      - uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
```

Comment:
This cache is... okay, but it's not doing as much work as it could... You'd probably benefit from also caching Yarn's cache, like the other jobs do.

Also - not sure how this deals with symlinks... I'd be tempted to be somewhat afraid it caches the referents of the symlinks to packages within the monorepo... Could this have consequences?

Reply:
Lemme know if this is more what you were thinking. Since this only needs yarn deps tried to exclude all the other languages package caches.